tools/text_processing/text_processing/awk.xml

```diff
@@ -15,15 +15,21 @@
             -v OFS='	'
             --re-interval
             -f "$awk_script"
-            "$infile"
+            #for group in $infiles
+                #for file in $group.infile
+                    "$file"
+                #end for
+            #end for
         > "$outfile"
 ]]>
     </command>
     <configfiles>
         <configfile name="awk_script">$code</configfile>
     </configfiles>
     <inputs>
-        <param name="infile" format="txt" type="data" label="File to process" />
+        <repeat name="infiles" title="Input files" min="1">
+            <param name="infile" format="txt" type="data" multiple="true" label="File to process" />
```

Comment:
The repeat could be removed correct?

Reply:
No, the repeat guarantees a file order, while multiple=true allows consuming a collection directly. Sorry, I totally forgot about running separately for each file. Let me make a change real quick..

Reply:
Uha, I see. Can you tell us the use-case. We might want to find a less hacky way here.

Reply:
Have a look at my other commit.